File: TimeValidator/TimeValidator.go
package main

import (
        "fmt"
        "strconv"
        )

func isValidTime(time string) bool {
    if len(time) != 4 {
        return false // Должно быть ровно 4 цифры
    }

    // Первые 2 символа - часы, последние 2 символа - минуты
    hours := time[:2]
    minutes := time[2:]

    // Преобразуем строки в числа
    h, err1 := strconv.Atoi(hours)
    m, err2 := strconv.Atoi(minutes)

    // Проверяем, что преобразование прошло успешно и диапазоны
    return err1 == nil && err2 == nil && h >= 0 && h <= 23 && m >= 0 && m <= 59
}

func main() {
    var input string
    fmt.Print("Введите последовательность чисел: ")
    fmt.Scan(&input)
    if isValidTime(input) {
        fmt.Println("YES")
    } else {
        fmt.Println("NO")
    }
}

File: TimeValidator/TimeValidator.rb
def valid_time?(time)
    # Должно быть ровно 4 цифры
return false if time.length != 4

    # Первые 2 символа - часы, последние 2 символа - минуты
    hours = time[0, 2]
minutes = time[2, 2]

    # Преобразуем строки в числа
h = hours.to_i
    m = minutes.to_i

# Проверяем диапазоны
    (0..23).include?(h) && (0..59).include?(m)
end

print "Введите последовательность чисел: "
input = gets.chomp

if valid_time?(input)
puts "YES"
    else
puts "NO"
end
